src/main/java/com/fasterxml/jackson/databind/jsontype/impl/ClassNameIdResolver.java

```diff
@@ -72,14 +109,21 @@ public JavaType typeFromId(DatabindContext context, String id) throws IOExceptio
 
     protected JavaType _typeFromId(String id, DatabindContext ctxt) throws IOException
     {
-        // 24-Apr-2019, tatu: [databind#2195] validate as well as resolve:
-        JavaType t = ctxt.resolveAndValidateSubType(_baseType, id, _subTypeValidator);
-        if (t == null) {
-            if (ctxt instanceof DeserializationContext) {
-                // First: we may have problem handlers that can deal with it?
-                return ((DeserializationContext) ctxt).handleUnknownTypeId(_baseType, id, this, "no such class found");
+        DeserializationContext deserializationContext = null;
+        if (ctxt instanceof DeserializationContext) {
+            deserializationContext = (DeserializationContext) ctxt;
+        }
+        if (_allowedSubtypes != null && deserializationContext != null
```

Comment:
Without `@JsonSubTypes`, wouldn't `_allowedSubtypes` end up being `null`... ?

Reply:
I guess in the current impl, you can omit the JsonSubTypes annotation and then all subtypes are valid. Maybe I should change that.

Reply:
@cowtowncoder I changed the code to enforce the check if enabled even when JsonSubTyoes are not defined